ContiguousArray.java

```diff
@@ -0,0 +1,43 @@
+import java.util.HashMap;
+import java.util.Map;
+
+
+//Time complexity : O(n) The complete array is traversed only once.
+//Space complexity :o(n)Maximum size of the HashMap map will be n, if all the elements are either 1 or 0.
+//Did this code successfully run on Leetcode : yes
+//Any problem you faced while coding this :no
+
+public class ContiguousArray {
+	public static int findMaxLength(int[] nums) {
+	//declare hashmap with key= count and value = current index
+    Map<Integer, Integer> map = new HashMap<>();
+    //initalize map with key =0 and value =-1
+    map.put(0, -1);
```

Comment:
Can you state the reason for doing so ?

Reply:
count 0 means no. of occurrences of 1 and 0 ares equal and which indicate got subarray with equal 1 and 0.  so to increase count by 1 for 0 keeping value  -1. As length =end index -(start index) so -1 become +1.